tests/test_ir.py

```diff
@@ -290,11 +293,6 @@ def test_dependences_eq(expr, expected, ti0, ti1, fa):
     (['Eq(ti0[x,y,z], ti1[x,y,z])',
       'Eq(ti3[x,y,z], ti0[y+1,y,y])'],
      ['ti0,flow,x', 'ti0,anti,x']),
-    # Data indices don't match iteration indices, so conservatively assume
```

Comment:
Note: I'm dropping this simply because it was conceptually broken. The new implementation spotted the flaw.